go.mod

```diff
@@ -40,6 +40,7 @@ require (
 	sigs.k8s.io/controller-runtime v0.6.3
 	sigs.k8s.io/controller-tools v0.4.1
 	sigs.k8s.io/yaml v1.2.0
+	github.com/cloudflare/backoff v0.0.0-20161212185259-647f3cdfc87a
```

Comment:
This library is available under [BSD-2-Clause License](https://github.com/cloudflare/backoff/blob/master/LICENSE). As far as I understand this licence, we can use it, but could also anyone else take a look at it?

Reply:
You can use the `Backoff` in `k8s.io/apimachinery/pkg/util/wait`.

Reply:
I considered this package, but the `Backoff` returns a fixed interval. There is also the `BackoffManger` interface which supports exponential backoff, but with different semantics. It returns the following channel `<-chan time.Time`, so it's not useful in our case.

Reply:
If you set the `Steps` and `Factor` fields properly, the interval increases every time:
```
	backoff := wait.Backoff{
		Duration: 10 * time.Millisecond,
		Factor:   2,
		Steps:    math.MaxInt32,
		Cap:      3 * time.Minute,
	}
	for i := 0; i < 20; i++ {
		fmt.Println(backoff.Step())
	}
```

The code above prints the following:
```
10ms
20ms
40ms
80ms
160ms
320ms
640ms
1.28s
2.56s
5.12s
10.24s
20.48s
40.96s
1m21.92s
2m43.84s
3m0s
3m0s
3m0s
```

Reply:
It seems that I misunderstood it, thanks. I will remove imported library and replace with it.

Reply:
Done.